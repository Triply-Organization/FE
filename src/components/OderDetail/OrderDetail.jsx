import { Col, Divider, Image, Row, Typography } from 'antd';
import { PropTypes } from 'prop-types';
import React from 'react';

import './OrderDetail.scss';

const { Title, Text } = Typography;

const OrderDetail = props => {
  const { data } = props;
  console.log(data);
  return (
    <>
      <Divider />
      <div className="checkout-info-display">
        <Row gutter={8}>
          <Col lg={8} md={6} sm={6} xs={0}>
            <Image
              width={150}
              src="https://demo2.pavothemes.com/triply/wp-content/uploads/2020/11/5c486ea739c52128a578315e_DSC04947-copy-150x150.jpg"
            />
          </Col>
          <Col lg={12} md={15} sm={15} xs={18}>
            <Title level={4}>{data.title}</Title>
            <section>
              <Text strong>Date: </Text>
              {data.date}
            </section>
            <Text strong>Duration: </Text>
            {data.duration} days
            <section>
              <Text strong>Ticket:</Text>
            </section>
            <section className="checkout-info-display__ticket-ctn">
              {data.tickets.adult.quantity && (
                <span>
                  <Text>Adult: </Text>
                  <Text strong>
                    ${data.tickets.adult.price} x {data.tickets.adult.quantity}
                  </Text>
                </span>
              )}
              {data.tickets.youth.quantity ? (
                <span>
                  <Text>Youth</Text>
                  <Text strong>
                    ${data.tickets.youth.price} x {data.tickets.youth.quantity}
                  </Text>
                </span>
              ) : null}
              {data.tickets.children.quantity ? (
                <span>
                  <Text>Children: </Text>
                  <Text strong>
                    ${data.tickets.children.price} x{' '}
                    {data.tickets.children.quantity}
                  </Text>
                </span>
              ) : null}
            </section>
          </Col>
          <Col lg={4} md={3} sm={3} xs={6}>
            <Title level={5}>
              {/* {data.tickets.adult.price +
                data.tickets.youth.price +
                data.tickets.children.price} */}
              $0.00
            </Title>
          </Col>
        </Row>
        <Divider />

        <Row gutter={8} className="ctn-result">
          <Col lg={8} md={6} sm={6} xs={0}></Col>
          <Col className="synthetic-price" lg={12} md={15} sm={15} xs={18}>
            <section>
              <Text strong>Subtotal</Text>
            </section>
            <section>
              <Text strong>Total</Text>
            </section>
            <section>
              <Text strong>Discount</Text>
            </section>
            <section>
              <Title level={5}>Amount Due</Title>
            </section>
          </Col>
          <Col lg={4} md={3} sm={3} xs={6}>
            <section>
              <Text strong>$0.00</Text>
            </section>
            <section>
              <Text strong>$0.00</Text>
            </section>
            <section>
              <Text strong>$0.00</Text>
            </section>
            <section>
              <Title level={5} className="result-payment">
                $0.00
              </Title>
            </section>
          </Col>
        </Row>
      </div>
    </>
  );
};

OrderDetail.propTypes = {
  data: PropTypes.object,
};

export default OrderDetail;
