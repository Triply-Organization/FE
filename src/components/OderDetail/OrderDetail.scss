.checkout-info-display {
    width: 100%;

    .ctn-result {
        background-color: #f6f6f6;
        padding: 2rem 0;

        .result-payment {
            color: #f7931e;
        }
    }

    &__ticket-ctn {
        display: flex;
        flex-direction: column;
    }

    .synthetic-price {
        text-align: right;
    }
}
