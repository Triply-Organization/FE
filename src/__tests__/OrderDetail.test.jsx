import {
  act,
  fireEvent,
  getByText,
  render,
  screen,
} from '@testing-library/react';
import { expect, it } from 'vitest';

import OrderDetail from '../components/OderDetail/OrderDetail';

describe('Orderdetail component', () => {
  it('should render correctly', () => {
    const result = render(<OrderDetail />);
    expect(result).toMatchSnapshot();
  });

  it('should render Text discount value', () => {
    const data = {
      duration: 4,
      id: 56,
      scheduleId: 2,
      startDay: {
        date: '2022-07-11 00:00:00.000000',
        timezone_type: 3,
        timezone: 'UTC',
      },
      status: 'unpaid',
      subTotal: 376,
      tax: null,
      tickets: {
        adult: { idTicket: 110, amount: 2, priceTick: 138 },
        children: { idTicket: 108, amount: 1, priceTick: 30 },
        youth: { idTicket: 109, amount: 1, priceTick: 70 },
      },
      tourId: 2,
      tourTitle: 'Go to Da Lat',
      user: 2,
      voucher: [
        { id: 1, code: 'VOUCHER', discount: 20, remain: 0 },
        { id: 2, code: 'VIPCODE', discount: 50, remain: 5 },
      ],
    };
    render(
      <OrderDetail
        discountValue={10}
        data={data}
        taxInfo={8}
        finalTotal={406.08}
      />,
    );
    expect(screen.getByTestId('text-discount').textContent).toBe(`-${10}%`);
  });

  it('should render bill discount number', () => {
    const data = {
      bill: {
        discount: '10',
      },
    };
    render(<OrderDetail data={data} />);
    expect(screen.getByTestId('text-discount').textContent).toBe('10');
  });

  it('should render bill discount number', () => {
    const finalTotal = 10;
    render(<OrderDetail finalTotal={finalTotal} />);
    expect(screen.getByTestId('text-finalTotal').textContent).toBe(
      finalTotal?.toLocaleString(`en-US`, {
        style: 'currency',
        currency: 'USD',
      }),
    );
  });

  // it('should render bill discount number', () => {
  //   const data = {
  //     imageTour: {
  //       path: 'https://car-rent-nhivo.s3.ap-southeast-1.amazonaws.com/upload/photo-1607356296477-5454f7a7f0f0-62c904df3523a.jpg',
  //     },
  //   };

  //   render(<OrderDetail data={data} />);
  //   expect(screen.getByTestId('test-imageUrl')).toContain(
  //     'https://car-rent-nhivo.s3.ap-southeast-1.amazonaws.com/upload/photo-1607356296477-5454f7a7f0f0-62c904df3523a.jpg',
  //   );
  // });

  it('should render Text discount value', () => {
    const data = {
      duration: 4,
      id: 56,
      scheduleId: 2,
      startDay: {
        date: '2022-07-11 00:00:00.000000',
        timezone_type: 3,
        timezone: 'UTC',
      },
      status: 'unpaid',
      subTotal: 376,
      tax: null,
      tickets: {
        adult: { idTicket: 110, amount: 2, priceTick: 138 },
        children: { idTicket: 108, amount: 1, priceTick: 30 },
        youth: { idTicket: 109, amount: 1, priceTick: 70 },
      },
      tourId: 2,
      tourTitle: 'Go to Da Lat',
      user: 2,
      voucher: [
        { id: 1, code: 'VOUCHER', discount: 20, remain: 0 },
        { id: 2, code: 'VIPCODE', discount: 50, remain: 5 },
      ],
    };
    render(
      <OrderDetail
        discountValue={10}
        data={data}
        taxInfo={8}
        finalTotal={null}
      />,
    );
    expect(screen.getByTestId('text-finalTotal').textContent).toBe(
      data?.subTotal?.toLocaleString(`en-US`, {
        style: 'currency',
        currency: 'USD',
      }),
    );
  });
});
