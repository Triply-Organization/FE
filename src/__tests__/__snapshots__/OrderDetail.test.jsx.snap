// Vitest Snapshot v1

exports[`Orderdetail component > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <div
        class="checkout-info-display"
      >
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-xs-0 ant-col-sm-6 ant-col-md-6 ant-col-lg-8"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-image"
              data-testid="test-imageUrl"
              style="width: 200px;"
            >
              <img
                class="ant-image-img"
              />
              <div
                class="ant-image-mask"
              >
                <div
                  class="ant-image-mask-info"
                >
                  <span
                    aria-label="eye"
                    class="anticon anticon-eye"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      />
                    </svg>
                  </span>
                  Preview
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col checkout-info-display__content ant-col-xs-22 ant-col-sm-18 ant-col-md-18 ant-col-lg-16"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <h4
              class="ant-typography"
            />
            <section>
              <span
                class="ant-typography"
              >
                <strong>
                  checkout.order_detail.date
                   
                </strong>
              </span>
              <span
                class="ant-typography"
              >
                2022-07-09
              </span>
            </section>
            <section>
              <span
                class="ant-typography"
              >
                <strong>
                  checkout.order_detail.time
                   
                </strong>
              </span>
              <span
                class="ant-typography"
              >
                12:00 pm
              </span>
            </section>
            <span
              class="ant-typography"
            >
              <strong>
                checkout.order_detail.duration
                 
              </strong>
            </span>
            <span
              class="ant-typography"
            >
               days
            </span>
            <section>
              <span
                class="ant-typography"
              >
                <strong>
                  checkout.order_detail.ticket.title
                   
                </strong>
              </span>
            </section>
            <section
              class="checkout-info-display__ticket-ctn"
            />
          </div>
        </div>
        <div
          class="ant-divider ant-divider-horizontal"
          role="separator"
        />
        <div
          class="ant-row ctn-result"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-xs-0 ant-col-sm-6 ant-col-md-6 ant-col-lg-8"
            style="padding-left: 4px; padding-right: 4px;"
          />
          <div
            class="ant-col synthetic-price ant-col-xs-18 ant-col-sm-15 ant-col-md-15 ant-col-lg-12"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <section>
              <span
                class="ant-typography"
              >
                <strong>
                  checkout.order_detail.sub_total
                </strong>
              </span>
            </section>
            <section>
              <span
                class="ant-typography"
              >
                <strong>
                  checkout.order_detail.discount
                </strong>
              </span>
            </section>
            <section>
              <span
                class="ant-typography"
              >
                <strong>
                  checkout.order_detail.tax
                </strong>
              </span>
            </section>
            <section>
              <h4
                class="ant-typography"
              >
                checkout.order_detail.total
              </h4>
            </section>
          </div>
          <div
            class="ant-col ant-col-xs-6 ant-col-sm-3 ant-col-md-3 ant-col-lg-3"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <section>
              <span
                class="ant-typography"
              >
                <strong />
              </span>
            </section>
            <section>
              <span
                class="ant-typography"
                data-testid="text-discount"
              >
                <strong />
              </span>
            </section>
            <section>
              <span
                class="ant-typography"
              >
                <strong>
                  8
                  %
                </strong>
              </span>
            </section>
            <section>
              <h4
                class="ant-typography result-payment"
                data-testid="text-finalTotal"
              />
            </section>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <div
      class="checkout-info-display"
    >
      <div
        class="ant-row"
        style="margin-left: -4px; margin-right: -4px;"
      >
        <div
          class="ant-col ant-col-xs-0 ant-col-sm-6 ant-col-md-6 ant-col-lg-8"
          style="padding-left: 4px; padding-right: 4px;"
        >
          <div
            class="ant-image"
            data-testid="test-imageUrl"
            style="width: 200px;"
          >
            <img
              class="ant-image-img"
            />
            <div
              class="ant-image-mask"
            >
              <div
                class="ant-image-mask-info"
              >
                <span
                  aria-label="eye"
                  class="anticon anticon-eye"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="eye"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                    />
                  </svg>
                </span>
                Preview
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-col checkout-info-display__content ant-col-xs-22 ant-col-sm-18 ant-col-md-18 ant-col-lg-16"
          style="padding-left: 4px; padding-right: 4px;"
        >
          <h4
            class="ant-typography"
          />
          <section>
            <span
              class="ant-typography"
            >
              <strong>
                checkout.order_detail.date
                 
              </strong>
            </span>
            <span
              class="ant-typography"
            >
              2022-07-09
            </span>
          </section>
          <section>
            <span
              class="ant-typography"
            >
              <strong>
                checkout.order_detail.time
                 
              </strong>
            </span>
            <span
              class="ant-typography"
            >
              12:00 pm
            </span>
          </section>
          <span
            class="ant-typography"
          >
            <strong>
              checkout.order_detail.duration
               
            </strong>
          </span>
          <span
            class="ant-typography"
          >
             days
          </span>
          <section>
            <span
              class="ant-typography"
            >
              <strong>
                checkout.order_detail.ticket.title
                 
              </strong>
            </span>
          </section>
          <section
            class="checkout-info-display__ticket-ctn"
          />
        </div>
      </div>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <div
        class="ant-row ctn-result"
        style="margin-left: -4px; margin-right: -4px;"
      >
        <div
          class="ant-col ant-col-xs-0 ant-col-sm-6 ant-col-md-6 ant-col-lg-8"
          style="padding-left: 4px; padding-right: 4px;"
        />
        <div
          class="ant-col synthetic-price ant-col-xs-18 ant-col-sm-15 ant-col-md-15 ant-col-lg-12"
          style="padding-left: 4px; padding-right: 4px;"
        >
          <section>
            <span
              class="ant-typography"
            >
              <strong>
                checkout.order_detail.sub_total
              </strong>
            </span>
          </section>
          <section>
            <span
              class="ant-typography"
            >
              <strong>
                checkout.order_detail.discount
              </strong>
            </span>
          </section>
          <section>
            <span
              class="ant-typography"
            >
              <strong>
                checkout.order_detail.tax
              </strong>
            </span>
          </section>
          <section>
            <h4
              class="ant-typography"
            >
              checkout.order_detail.total
            </h4>
          </section>
        </div>
        <div
          class="ant-col ant-col-xs-6 ant-col-sm-3 ant-col-md-3 ant-col-lg-3"
          style="padding-left: 4px; padding-right: 4px;"
        >
          <section>
            <span
              class="ant-typography"
            >
              <strong />
            </span>
          </section>
          <section>
            <span
              class="ant-typography"
              data-testid="text-discount"
            >
              <strong />
            </span>
          </section>
          <section>
            <span
              class="ant-typography"
            >
              <strong>
                8
                %
              </strong>
            </span>
          </section>
          <section>
            <h4
              class="ant-typography result-payment"
              data-testid="text-finalTotal"
            />
          </section>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
